### DISCUSSION TITLE
Determining TCR from nuclei unspliced RNA

### DISCUSSION THREAD

**Discussion by racng** [milaboratory/mixcr#1515] [Q&A] [answered]
Can MiXCR perform VDJ assembly on nuclei-derived RNA that would contain mostly unspliced transcripts? What preset would be appropriate? Thanks!

**Comment by mizraelson** [comment]
Hi, it certainly can. What is the library structure? Is there a published protocol?

**Reply by racng** [reply]
It would be using the 10x 5' VDJ kit. According to conversation with 10x support in June 2023, they had concerns that "it's unclear how the unspliced transcripts may change the mapping to the reference and make it more difficult for the algorithm to assemble the VDJ sequences", at least for their own cellranger vdj software. I was wondering if mixcr can overcome this potential problem. 

Are there any introns in the CDR3 region? I see that usually there is an intron near the beginning of the V gene according to your [documentation](https://mixcr.com/mixcr/reference/ref-gene-features/). Does MiXCR deal with that in their alignment algorithm?

**Reply by mizraelson** [reply, marked as answer]
So there are two sites that undergo splicing:
-  the first one is intron in the leader sequence at the beginning of V gene
-  the second one is the region that includes all non recombined J genes to the right of the VDJ junction up to the beginning of the C gene.

The V gene intron shouldn't be a problem, if you specify `--dna` parameter MiXCR will use a DNA reference which includes this intron. There might be an issue with C gene (isotype in case of B cells) because this part (between J and C genes in pre-mRNA) of the transcript varies depending on the particular J gene used. Will you be interested in full-length sequences or just the CDR3?

**Reply by racng** [reply]
I am interested in the determination of at least the CDR3, V gene, and J gene of alpha and beta chain in T cells . Ideally enough information to reconstruct the whole TCR sequence, which means also know the C gene?

**Reply by mizraelson** [reply]
Since the C gene introduces limited diversity in TCRs, with only one TRAC and two TRBC genes, it's typically not essential to focus on it. While it will be still possible to recover the C gene, this might require extra effort in examining alignments. Identifying the rest of the TCR sequence should be straightforward.
